src/frontend/src/handler/query.rs

```diff
@@ -29,7 +32,7 @@ use crate::scheduler::{
 };
 use crate::session::{OptimizerContext, SessionImpl};
 
-pub type QueryResultSet = Vec<Row>;
+pub type QueryResultSet = BoxStream<'static, std::result::Result<Row, BoxedError>>;
```

Comment:
Can we use generic to avoid this `Box`? There could be millions of rows in the result set, and I'm not sure whether this impacts the performance a lot.

Reply:
Indeed it may impact the performance. I try to use generic but it will make so many places full of generic param. I think it's not appropriate to use generic param( at least the way I use .
I think there are other ways to solve this problem:
- use the type implement stream trait to avoid generic param 
- use Vec<Row> instead of Row to decrease the time of call next()
  BoxStream<'static, std::result::Result<Vec<Row>,BoxedError>>

Reply:
> Indeed it may impact the performance.

Could you elaborate this point? 👀

Reply:
Oh yeees, I misread it as "may not impact" 😄😄😄

Reply:
> use the type implement stream trait to avoid generic param

Yes. We may Type Alias Impl Trait here.

Reply:
> use Vec instead of Row to decrease the time of call next()
> BoxStream<'static, std::result::Result<Vec,BoxedError>>

I have used this way to decrease the time of calling stream.next() .
And I will create a new PR to create a new stream type to avoid Boxstream later.

Reply:
Please create an issue for that